presto-main/src/main/java/io/prestosql/server/ui/FormWebUiAuthenticationManager.java

```diff
@@ -318,21 +319,10 @@ private static String getRedirectLocation(HttpServletRequest request, String pat
 
     static String getRedirectLocation(HttpServletRequest request, String path, String queryParameter)
     {
-        HttpUriBuilder builder;
-        if (isNullOrEmpty(request.getHeader(X_FORWARDED_HOST))) {
-            // not forwarded
-            builder = uriBuilder()
-                    .scheme(request.getScheme())
-                    .host(request.getServerName())
-                    .port(request.getServerPort());
-        }
-        else {
-            // forwarded
-            builder = uriBuilder()
+        HttpUriBuilder builder = uriBuilder()
                     .scheme(firstNonNull(emptyToNull(request.getHeader(X_FORWARDED_PROTO)), request.getScheme()))
-                    .host(request.getHeader(X_FORWARDED_HOST));
-            getForwarderPort(request).ifPresent(builder::port);
-        }
+                    .host(firstNonNull(emptyToNull(request.getHeader(X_FORWARDED_HOST)), request.getServerName()))
+                    .port(getForwarderPort(request).orElse(request.getServerPort()));
```

Comment:
If there is
```
X-Forwarder-Proto https
X-Forwarder-Host my-load-balancer.local
```

then using 8080 (in absence of `X-Forwarded-Port`) seems unreasonable.
Using https' default would make more sense to me. 
(This is why i wrote the code this way.)

@electrum?

Reply:
I tried looking at the spec, but it doesn't say anything useful. So I suggest we just make sure that AWS, GCP, and Azue LBs work:
* ALB: https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/x-forwarded-headers.html
* GCP: https://cloud.google.com/load-balancing/docs/https
* Asure: https://docs.microsoft.com/bs-latn-ba/azure/frontdoor/front-door-http-headers-protocol
One thing interesting, it looks like the `X-Forwarded-Host` header can contain a port.

Reply:
Interesting. Seems AWS ALB, GCP don't document `X-Forwarded-Host` at all, so if we want to support that, we need to find out the load balancer IP address differently. For example, GCP suggests it puts a bunch of different addresses in `X-Forwarded-For` and one _could be_ the load balancer's IP.

... I am probably missing something obvious, this doesn't sound convincing to myself.

Reply:
Hi,

We are facing same problem as well at Alicloud SLB which does not support `X-Forwarded-Host`. Just provides `X-Forwarded-For` and `X-Forwarded-Proto`. Somehow `https` should be activated when `X-Forwarded-Proto` present.

BTW, i have tried this PR, but situation seems persistent.

Reply:
@findepi and @veysiertekin I don't think they provide `X-Forwarded-Host` because it isn't needed.  For those systems, the host is unchanged from the host header in the request, because the LB run on the same host just on a different port.

Reply:
@dain  in AWS ALB cases, the host (machine) is definitely separate. But, maybe, we get it in the `Host` header and that's why we don't need it?

Reply:
From the original issue, his headers are:
```
X-Forwarded-Proto: https
X-Forwarded-Port: 443
Host: presto.domain.net
```
In this case, we change the protocol, and port, but not the host which was `presto.domain.net`.